packages/frontend/editor-ui/src/components/WhatsNewModal.vue

```diff
@@ -47,25 +47,21 @@ const onUpdateClick = async () => {
 	await pageRedirectionHelper.goToVersions();
 };
 
-watch(
-	() => versionsStore.whatsNewArticles,
-	(whatsNewArticles) => {
-		for (const article of whatsNewArticles) {
-			if (!versionsStore.isWhatsNewArticleRead(article.id)) {
-				versionsStore.setWhatsNewArticleRead(article.id);
-			}
-		}
-	},
-	{ immediate: true },
-);
-
 modalBus.on('opened', () => {
 	const articleIndex = versionsStore.whatsNewArticles.findIndex(
 		(article) => article.id === props.data.articleId,
 	);
 
 	scroller.value?.scrollToItem(articleIndex);
 });
+
+onMounted(() => {
```

Comment:
This looks better, but this logic doesn't really has to do anything with the components, it just updates the state in the store based on the state in the store. So I would suggest moving it to the versions store and setting read articles after they are fetched in the `checkForNewVersions` method. We can also create `init` method in versions store that does these things (potentially other init logic) and is called from the `init.ts`